test/route-basePath.js

```diff
@@ -48,13 +48,22 @@ describe('PushStateTree Router basePath', function () {
     })).basePath).to.equal('/folder/file.ext?param=value');
   });
 
-  it('should use the non relative root as basePath if not specified', function () {
-    history.pushState(null, null, '/abc/123/');
+  it('should convert empty to slash', () => {
     var pst = new PushStateTree();
     expect(pst.basePath).to.equal('/');
   });
 
-  it('should not share the basePath between route instances', function () {
+  it('should detect when ends without slash', () => {
+    var pst = new PushStateTree({
+      basePath: 'file.html'
+    });
+    history.pushState(null, null, '/file.html/test');
+    //TODO: history.pushState is not dispatching popstate event
```

Comment:
I couldn't figure out yet how to fix this correctly, I'm not sure, but looks like to be related to the test lib, or PhantomJS, because it never happen on other browsers and in the manual testes I couldn't reproduce this bug yet.

We need a way to reproduce it, create a test-case that cover the problem on not triggering the `popstate` event then we might be able to fix it.

One of my ideas to go around such bug (and it's a WIP) is to separate the `popstate` or `hashchange` from the lib as dependency and have a exposed method that trigger the internal router processing, them we can have a separated test to only test browser interactions such dispatching this method when popstate, hashchange, load and other events that should start the router react.